dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/GoogleHelper.java

```diff
@@ -152,7 +160,30 @@ static boolean isValidAudience(Tokeninfo tokenInfo) {
     }
 
     private static boolean isValidToken(String googleToken) {
-        return tokenInfoFromToken(googleToken).map(tokenInfo -> isValidAudience(tokenInfo)).orElse(false);
+        return tokenInfoFromToken(googleToken).map(GoogleHelper::isValidAudience).orElse(false);
+    }
+
+    /**
+     * Gets the Google TokenResponse
+     *
+     * @param googleClientID
+     * @param googleClientSecret
+     * @param code        The satellizer code
+     * @param redirectUri The Google redirectUri
+     * @return
+     */
+    public static TokenResponse getTokenResponse(String googleClientID, String googleClientSecret, String code, String redirectUri) {
+        final AuthorizationCodeFlow flow = new AuthorizationCodeFlow.Builder(BearerToken.authorizationHeaderAccessMethod(), TokenResource.HTTP_TRANSPORT,
+            TokenResource.JSON_FACTORY, new GenericUrl(GoogleHelper.GOOGLE_ENCODED_URL),
+            new ClientParametersAuthentication(googleClientID, googleClientSecret), googleClientID,
+            GoogleHelper.GOOGLE_AUTHORIZATION_SERVICE_ENCODED_URL).build();
+        try {
+            return flow.newTokenRequest(code).setRedirectUri(redirectUri)
+                .setRequestInitializer(request -> request.getHeaders().setAccept("application/json")).execute();
+        } catch (IOException e) {
+            LOG.error("Retrieving accessToken was unsuccessful");
+            throw new CustomWebApplicationException("Could not retrieve google.com token based on code", HttpStatus.SC_BAD_REQUEST);
```

Comment:
This is super minor, but I would not say `google.com`, as you will typically be logging in with another domain, e.g., I log in with a ucsc.edu domain. Maybe just `google token`.